File: src/main/java/net/shadowfacts/craftingslabs/compat/craftingtweaks/CompatCraftingTweaks.java
package net.shadowfacts.craftingslabs.compat.craftingtweaks;

import net.blay09.mods.craftingtweaks.api.CraftingTweaksAPI;
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
import net.shadowfacts.craftingslabs.container.ContainerCrafting;
import net.shadowfacts.shadowmc.compat.Compat;

/**
 * @author shadowfacts
 */
@Compat("craftingtweaks")
public class CompatCraftingTweaks {

	@Compat.PostInit
	public static void postInit(FMLPostInitializationEvent event) {
		CraftingTweaksAPI.registerProvider(ContainerCrafting.class, new CSTweakProvider());
	}

}

File: src/main/kotlin/net/shadowfacts/craftingslabs/compat/CompatCraftingTweaks.kt
package net.shadowfacts.craftingslabs.compat

import net.minecraft.nbt.NBTTagCompound
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLInterModComms
import net.shadowfacts.craftingslabs.container.ContainerCrafting
import net.shadowfacts.shadowmc.compat.Compat

/**
 * @author shadowfacts
 */
@Compat("craftingtweaks")
object CompatCraftingTweaks {

	@JvmStatic
	@Compat.Init
	fun init(event: FMLInitializationEvent) {
		FMLInterModComms.sendMessage("craftingtweaks", "RegisterProvider", NBTTagCompound().apply {
			setString("ContainerClass", ContainerCrafting::class.qualifiedName)
			setInteger("GridSlotNumber", 1)
			setInteger("GridSize", 9)
			setString("AlignToGrid", "left")
		})
	}

}